### PR TITLE
Only fetch collections when they're open

### PR DESCRIPTION
**Pull request by jonathonherbert**
Does pretty much what it says on the tin -- only fetch collections when they're opened. This might be when the front loads, or it might be when a user opens a previously unopened collection.

We always fetch a new collection on open, as its contents may have changed (polling will solve the problem of long-opened collections). We could have a staleness check or debounce if this causes problems, for example if a user rapidly toggles a collection open and closed.

The large file count is partly due to the additional files for fixtures -- I got a bit carried away scaffolding testing the 'initialiseFront' thunk! Sorry all.

### CODE REVIEW

File: client-v2/src/util/configureStore.ts [hunk 1]
```diff
@@ -15,25 +11,26 @@ import {
   persistClipboardOnEdit,
   persistOpenFrontsOnEdit
 } from './storeMiddleware';
+import { State } from 'types/State';
 
-export default function configureStore() {
+export default function configureStore(initialState: State) {
```
Comment (jonathonherbert) [inline client-v2/src/util/configureStore.ts]: in tests, this lets us inject fixture state into the store as it's used in the application, for end-to-end thunk testing. We've used mock-store previously, but I think I'm beginning to prefer this approach as it's as close to the in-app setup as possible. Interested in any thoughts 😄

File: client-v2/src/types/Config.ts [hunk 2]
```diff
@@ -24,8 +24,7 @@ interface Config {
   sentryPublicDSN: string,
   mediaBaseUrl: string,
   apiBaseUrl: string,
-  stage: string,
-  switches: unknown,
+  switches: {[key: string]: boolean},
```
Comment (jonathonherbert) [inline client-v2/src/types/Config.ts]: The type here was previously `unknown`, which was breaking store initialisation with state -- this is closer to the type as it stands IRL.

File: client-v2/src/actions/Collections.ts [hunk 3]
```diff
@@ -35,69 +35,88 @@ import { getCollectionConfig } from 'selectors/frontsSelectors';
...
-        const {
-          normalisedCollection,
-          articleFragments,
-          groups
-        } = normaliseCollectionWithNestedArticles(
-          collectionWithDraftArticles,
-          collectionConfig
-        );
+      const collections = await fetchCollection(collectionIds);
+      await Promise.all(
```
Comment (RichieAHB) [inline client-v2/src/actions/Collections.ts]: Is the only change in this area the removal of `flatten` 😨

**Comment by jonathonherbert** [comment]
Thanks @RichieAHB, yes, as discussed something to enforce a `prettier` pass would be ideal 👍

**Comment by RichieAHB** [comment]
@jonathonherbert just realised that this also trriggers an update on any Overview click - hadn't appreciated that at the time and perhaps you had but I thought it was worth calling out.

**Pull request merged**